src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java

```diff
@@ -116,6 +129,35 @@ public ClientResponseModel joinChannel(ChannelOperationRequest channelOperationR
     }
   }
 
+  @Override
+  public Set<String> getChannelMembersMSPID(String channelName) {
+
+    Channel channel = hfClientWrapper.getHfClient().getChannel(channelName);
+    if (channel == null) {
+      Network network = gateway.getNetwork(channelName);
+      Channel networkChannel = network.getChannel();
+      try {
+        Channel newChannel = hfClientWrapper.getHfClient().deSerializeChannel(networkChannel.serializeChannel());
+        return new HashSet<>(newChannel.getPeersOrganizationMSPIDs());
+      } catch (Exception e) {
+        log.error("Error in fetching channel members", e);
+        throw new ChannelOperationException(ErrorCode.CHANNEL_MEMBERS_MSPID_FAILED, "get channel members mspid failed",
+            e.getCause());
+      }
+    }
+
+    Configtx.Config channelConfig;
+    try {
+      final byte[] channelConfigurationBytes = channel.getChannelConfigurationBytes();
+      channelConfig = Configtx.Config.parseFrom(channelConfigurationBytes);
+    } catch (Exception e) {
+      log.error("Error in fetching channel members", e);
+      throw new ChannelOperationException(ErrorCode.CHANNEL_MEMBERS_MSPID_FAILED, "get channel members mspid failed",
```

Comment:
Error message should be more appropriate. As here performed operation is different.